### DISCUSSION TITLE
Apply patches from non-root packages

### DISCUSSION THREAD

**Discussion by shineability** [cweagans/composer-patches#540] [Q&A] [answered]
We have a Laravel package `vendor/laravel-azure-blob-storage` that depends on a package in which some files need to be patched. Setting everything up in the package's `composer.json` and running `composer install` works fine when the package is the root. However, when we require this package as dependency in our applications' `composer.json`, the patches are **not** applied. 

Is this by design and are patches only applied when configured in a root package or am I missing something?

**Comment by cweagans** [comment]
If you're on 1.x, see the readme heading "Allowing patches to be applied from dependencies".

If you're on 2.x, see https://www.cweagans.net/2023/07/dependency-patch-resolution/ heading "Moving forward".

**Comment by shineability** [comment]
@cweagans 

Thanks for the quick reply! And my bad, I failed to look for the 1.x docs in the right branch :/

I'm able to apply patches from my dependency now by setting the `enable-patching` flag. I am however, running into problems with the paths of the patch files now...

When patching from a dependency the path needs to be prefixed with the `vendor` folder in order to work, when working on the package as a root (running tests), it doesn't...

Is there a way to make it work in both scenarios without having to change the `composer.json`?

```
"extra": {
        "patches": {
            "microsoft/azure-storage-common": [
                "vendor/company/laravel-azure-blob-storage/patches/from-a-dependency.patch",
                "patches/from-a-dependency.patch",
            ],
            
```

**Reply by cweagans** [reply, marked as answer]
Nope, sorry. Upload the patch somewhere and use the URL instead.

**Reply by shineability** [reply]
Good enough, thanks for your feedback :)
